src/scripts/components/VariableMessage.jsx

```diff
@@ -328,37 +424,189 @@ class VariableMessage extends React.Component {
             variable={v}
             key={v.id}
             onClick={this.handleVariableSelection}
+            draggable={isDraggable}
+            id={v.value}
+            onDragStart={this.onVariableDragStart}
           >
             {v.value}
           </ToggleButton>
         );
       })
   )
 
-  render() {
+  changeCategoryHandler = (category) => {
+    this.setState({
+      variablesOfCategory: this.props.data.filter(item => item.category === category),
+      selectedCategory: category,
+    }, () => this.handleInitValue(this.state.message));
+  }
+
+  categoryMobileView = () => {
+    const {
+      characterCountTitle,
+      variableExplanationMessage,
+      showCharacterCounter,
+      characterCountWarningLength,
+    } = this.props;
+    const {
+      message,
+      variablesOfCategory,
+      selectedCategory,
+    } = this.state;
+    const characterCounterClasses = cx('variable-message__character-count', {
+      'variable-message__character-count--warning': message.length >= characterCountWarningLength,
+    });
+
+    return (
+      <div className="category-mobile-view">
+        <FormLabel className="u-block u-m-t" id="Variables">Variables</FormLabel>
+        <div className="category-dropdown">
+          <Dropdown label={selectedCategory} wide block>
+            {this.state.categories.map(item => (
+              <DropdownMenuItem
+                key={item}
+                id={item}
+                label={item}
+                onClick={() => this.changeCategoryHandler(item)}
+              />))}
+          </Dropdown>
+        </div>
+        <div className="category-variable-list u-m-y u-p-a">
+          {variableExplanationMessage && (
+            <div className="u-text-muted u-text-small">{variableExplanationMessage}</div>
+          )}
+          <div className="category-message__footer__variable__list">
+            {this.renderToggleButtons(variablesOfCategory)}
+          </div>
+        </div>
+        {showCharacterCounter && (
+          <div title={characterCountTitle} className={characterCounterClasses}>
+            {message.length}
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  categoryWebView = () => {
```

Comment:
same as above. Although I'm not sure how this will change if this is handled via CSS media queries. It would be preferable as stated above since it creates a better user experience but also gets rid of a lot of JSX code duplication from these render methods.